File: Prueba y error/ejercicios de semana santa php/Two php/Library/Plan.php
<?php
namespace Library;

abstract class Plan {
    protected string $nombre;
    protected int $costo;
    protected string $servicios;

    public function __construct(string $nombre, int $costo, string $servicios) {
        $this->nombre = $nombre;
        $this->costo = $costo;
        $this->servicios = $servicios;
    }

    public function getNombre(): string {
        return $this->nombre;
    }

    public function getCosto(): int {
        return $this->costo;
    }

    public function getServicios(): string {
        return $this->servicios;
    }
}

File: Prueba y error/ejercicios de semana santa java/two java/Library/Plan.java
package Library;

public abstract class Plan {
    private String nombre;
    private String servicios;
    private double precio;

    public Plan(String nombre, String servicios, double precio) {
        this.nombre = nombre;
        this.servicios = servicios;
        this.precio = precio;
    }

    public String getNombre() {
        return nombre;
    }

    public String getServicios() {
        return servicios;
    }

    public double getPrecio() {
        return precio;
    }

    // Método abstracto para obtener detalles específicos de cada plan
    public abstract String detalles();
}
